### PR TITLE
Strandsort

### LINKED ISSUE #303: [Feature Request] <Shell sort>
- ## 🚀 Feature Request
- 
- ### Is your feature request related to a problem? Please describe.
-I often find myself needing a more efficient sorting algorithm for datasets that are partially sorted or when performance is critical. Shell Sort is an advanced version of the insertion sort algorithm that allows for the exchange of items that are far apart. The primary goal of Shell Sort is to improve the efficiency of insertion sort by allowing the comparison and exchange of elements that are farther apart, thereby reducing the number of inversions in the array.
- 
- ### Describe the solution you'd like
- I would like to implement the Shell Sort algorithm as a sorting option within our project. Shell Sort is an efficient, generalized version of insertion sort that works by comparing elements at a certain gap and then reducing the gap until it becomes one. This allows the algorithm to efficiently move elements into their correct positions, making it much faster than regular insertion sort for larger datasets.
- 
- ### Describe alternatives you've considered
- 
- 
- ### Additional context
- Add any other context or screenshots about the feature request here.
- 
- ---
- 
- **Would you like to work on this feature?**  
- - [x] Yes  
- - [ ] No

### PR DESCRIPTION
**Pull request by priyashuu**
### Description
shell sort was already there  , so i add strand sort 
Strand Sort is a recursive, comparison-based sorting algorithm that works by pulling out sorted sublists (strands) from the original list and then merging them together to create the final sorted list. Strand Sort works best with linked lists due to its insertions and extraction steps but can also be implemented with arrays.

Fixes #303 

### Type of change

- [] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [] Documentation update

### Checklist:

- [x] My code follows the style guidelines of this project
- [x] I have performed a self-review of my code
- [x] I have commented my code, particularly in hard-to-understand areas
- [x] I have added tests that prove my fix is effective or that my feature works
- [x] New and existing unit tests pass locally with my changes
- [x] Any dependent changes have been merged and published in downstream modules

image

### CODE REVIEW

File: docs/algorithms/sorting-algorithms/OddEvenSort.md [hunk 1]
```diff
@@ -0,0 +1,99 @@
+
```
Comment (ajay-dhangar) [inline docs/algorithms/sorting-algorithms/OddEvenSort.md:1]: We need on top

```md
---
id: ___
title: ___
sidebar_label: ___
sidebar_position: ___
description: ___
tags: [___, __, ]
---
```

### Detailed Guidelines

- id: Unique identifier for the route.
- title: Title of the problem.
- sidebar_label: Problem number or label for the sidebar.
- tags: Relevant tags for the problem.
- description: A short SEO-friendly description. No HTML tags.

**Review by ajay-dhangar — CHANGES_REQUESTED** [review]
update your code based on suggestions

**Comment by priyashuu** [comment]
image

**Comment by priyashuu** [comment]
done

**Comment by priyashuu** [comment]
@ajay-dhangar done

**Comment by priyashuu** [comment]
@ajay-dhangar done with the changes

**Comment by priyashuu** [comment]
@ajay-dhangar done with the changes

**Comment by ajay-dhangar** [comment]
> @ajay-dhangar done with the changes

Please commit and push your updated code to this PR. Once you have completed your changes, please let me know so I can review and merge your changes.

**Comment by priyashuu** [comment]
@ajay-dhangar  done sir

**Review by ajay-dhangar — APPROVED** [review]
...

**Pull request merged**